### PR TITLE
Add unit tests for builders #543

### LINKED ISSUE #543: Cover summary builders with unit tests
**Description**

We need to cover  `SimpleCommentBuilder.kt`, `SimpleNameBuilder.kt`, and `SimpleClusterBuilder.kt` with unit tests to capture the current behavior and make it easier to catch issues in the future.

I will try to do it using UTBot and on my own as well.

### PR DESCRIPTION
**Pull request by onewhl**
# Description

Added unit tests for SimpleNameBuilder, SimpleCommentBuilder, SimpleClusterCommentBuilder.
Unfortunately, most of the methods are private and kotlin properties not always could be mocked, so the number of tests is small.

Fixes #543 

## Type of Change

Please delete options that are not relevant.

- New tests

# How Has This Been Tested?

## Automated Testing

Run tests in PR.

# Checklist (remove irrelevant options):

_This is the author self-check list_

- [ ] The change followed the style guidelines of the UTBot project
- [x] Self-review of the code is passed
- [ ] The change contains enough commentaries, particularly in hard-to-understand areas
- [ ] New documentation is provided or existed one is altered
- [x] No new warnings
- [x] New tests have been added
- [x] All tests pass locally with my changes

### CODE REVIEW

File: utbot-summary/build.gradle [hunk 1]
```diff
@@ -9,5 +9,7 @@ dependencies {
 
     implementation group: 'com.github.javaparser', name: 'javaparser-core', version: '3.22.1'
 
+    testImplementation 'org.mockito:mockito-core:4.7.0'
```
Comment (amandelpie) [inline utbot-summary/build.gradle:12]: There are a few mockito versions in the framework, to have less, I suggest to make the same thing like in the testImplementation below  `testImplementation("org.junit.jupiter:junit-jupiter:$junit5_version")` using $mockito_version, please search in the project

File: utbot-summary/src/test/kotlin/org/utbot/summary/comment/SimpleClusterCommentBuilderTest.kt [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        `when`(step.stmt).thenReturn(statement)
+        `when`(traceTag.path).thenReturn(listOf(step))
+        `when`(traceTag.rootStatementTag).thenReturn(statementTag)
+        `when`(traceTag.result).thenReturn(UtOverflowFailure(Throwable()))
+
+        sootToAst[sootMethod] = jimpleToASTMap
+    }
+
+    @Test
+    fun `throws throwable if execution result is null`() {
```
Comment (amandelpie) [inline utbot-summary/src/test/kotlin/org/utbot/summary/comment/SimpleClusterCommentBuilderTest.kt:48]: This test fails on my side with 
```
expected: <<pre>
Test throws Throwable 
</pre>> but was: < >
```

File: utbot-summary/src/test/kotlin/org/utbot/summary/comment/SimpleClusterCommentBuilderTest.kt [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import org.utbot.framework.plugin.api.UtOverflowFailure
+import org.utbot.summary.ast.JimpleToASTMap
+import org.utbot.summary.tag.StatementTag
+import org.utbot.summary.tag.TraceTag
+import soot.SootMethod
+import soot.jimple.Stmt
+import soot.jimple.internal.JReturnStmt
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class SimpleClusterCommentBuilderTest {
```
Comment (amandelpie) [inline utbot-summary/src/test/kotlin/org/utbot/summary/comment/SimpleClusterCommentBuilderTest.kt:19]: Could we add a few tests to cover the following method:
- buildDocStmts

File: utbot-summary/src/test/kotlin/org/utbot/summary/comment/SimpleCommentBuilderTest.kt [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+import org.utbot.framework.plugin.api.UtOverflowFailure
+import org.utbot.summary.ast.JimpleToASTMap
+import org.utbot.summary.tag.StatementTag
+import org.utbot.summary.tag.TraceTag
+import soot.SootMethod
+import soot.jimple.Stmt
+import soot.jimple.internal.JReturnStmt
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class SimpleCommentBuilderTest {
```
Comment (amandelpie) [inline utbot-summary/src/test/kotlin/org/utbot/summary/comment/SimpleCommentBuilderTest.kt:19]: Could we add a few tests to cover the following method:

- buildDocStmts

**Review by amandelpie — CHANGES_REQUESTED** [review]
It's a good question, should we test private methods or not, I suggest to use reflection or change the visibility of a few methods to the package-private or internal and test private methods without mockito but after moving from Soot to JCDB

If you wish, you could play with changing visibility at this PR for 1-2 method for the SimpleNameBuilder

Please add a few tests if possible and fix the issue with the mockito verision

File: utbot-summary/build.gradle (hunk 1 above)
Comment (onewhl) [reply utbot-summary/build.gradle]: I fixed it, thank you!

File: utbot-summary/src/test/kotlin/org/utbot/summary/comment/SimpleClusterCommentBuilderTest.kt (hunk 2 above)
Comment (onewhl) [reply utbot-summary/src/test/kotlin/org/utbot/summary/comment/SimpleClusterCommentBuilderTest.kt]: i didn't expect these two builders have different behavior, but now we know. I fixed test

**Comment by amandelpie** [comment]
@onewhl LGTM for me

**Review by amandelpie — APPROVED** [review]

**Pull request merged**